common/src/main/java/org/apache/drill/common/util/DrillPlatformDependent.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.common.util;
+
+import io.netty.util.internal.PlatformDependent;
+
+import java.lang.reflect.Method;
+
+public class DrillPlatformDependent {
+  private final static long MAX_DIRECT_MEMORY = maxDirectMemory0();
+
+  public static long maxDirectMemory() {
+    return MAX_DIRECT_MEMORY;
+  }
+
+
+  private static long maxDirectMemory0() {
```

Comment:
Why method is named with 0 in the end?

Reply:
Just a copy-paste from netty PlatformDependent.

Reply:
1. Please add javadoc for this class explaining its purpose.
2. I still suggest to rename the method.
3. Maybe add trace for the ignored exception, just in case we might need to find out what has happened.

Reply:
Why is copy/paste necessary (why not to use netty `PlatformDependent` directly)?

Reply:
netty PlatformDependent do not try to access jdk.internal.misc.VM.
By "copy-paste" i mean only a method name

Reply:
Looks like it can cause OOM:
In Drill version of `PooledByteBufAllocatorL` exists a reflective access to `PooledByteBufAllocator.directArenas` field. It's size depends on 
` DEFAULT_NUM_DIRECT_ARENA = Math.max(0,
                SystemPropertyUtil.getInt(
                        "io.netty.allocator.numDirectArenas",
                        (int) Math.min(
                                defaultMinNumArena,
                                PlatformDependent.maxDirectMemory() / defaultChunkSize / 2 / 3)));`

Reply:
Drill delegates to netty all large memory allocation and netty internally tracks how much direct memory is available. If netty incorrectly determines max direct memory size or Drill and netty max direct memory sizes are different, it is easy to get into a situation where netty will fail allocation when memory is available or it will try to allocate when memory is not available.

Reply:
netty `PlatformDependent` and `jdk.internal.misc.VM` returns same value for maxDirectMemory, so i remove my custom class